sdk/cosmos/azure-cosmos/azure/cosmos/_execution_context/non_streaming_order_by_aggregator.py

```diff
@@ -0,0 +1,168 @@
+# The MIT License (MIT)
+# Copyright (c) 2024 Microsoft Corporation
+
+"""Internal class for multi execution context aggregator implementation in the Azure Cosmos database service.
+"""
+
+import heapq
+from azure.cosmos._execution_context.base_execution_context import _QueryExecutionContextBase
+from azure.cosmos._execution_context import document_producer
+from azure.cosmos._routing import routing_range
+from azure.cosmos import exceptions
+
+
+# pylint: disable=protected-access
+
+class FixedSizePriorityQueue:
+    """Provides a Fixed Size Priority Queue abstraction data structure"""
+
+    def __init__(self, max_size):
+        self._heap = []
+        self.max_size = max_size
+
+    def pop(self):
+        return heapq.heappop(self._heap)
+
+    def push(self, item):
+        if len(self._heap) < self.max_size:
+            heapq.heappush(self._heap, item)
+        else:
+            heapq.heappushpop(self._heap, item)
+
+    def peek(self):
+        return self._heap[0]
+
+    def size(self):
+        return len(self._heap)
+
+
+class _NonStreamingOrderByContextAggregator(_QueryExecutionContextBase):
```

Comment:
>_NonStreamingOrderByContextAggregator [](http://example.com/codeflow?start=6&length=37)

WOuld it be possible to share code with `_MultiExecutionContextAggregator` instead of duplicating everything here ?

Reply:
while a lot of the code is similar there are some small differences that make it so the methods can't be directly inherited from the `_MultiExecutionContextAggregator` without having to change a lot of other things there as well